packages/gasket-plugin-https/index.d.ts

```diff
@@ -0,0 +1,71 @@
+import type { GasketConfig, HookExecTypes, MaybeMultiple, MaybeAsync } from '@gasket/engine';
+import type { SecureContextOptions } from 'tls';
+import type { Server as HttpServer } from 'http';
+import type { Server as HttpsServer } from 'https';
+import type { SecureServerOptions, Http2Server } from 'http2';
+import type { TerminusOptions, HealthCheckError } from '@godaddy/terminus';
+
+declare module '@gasket/engine' {
+  type BaseListenerConfig = {
+    port?: number,
+    host?: string,
+    timeout?: number
+  }
+
+  type CertInput = string | Buffer;
+
+  type CustomHttpsSettings = {
+    root?: string,
+    key: MaybeMultiple<CertInput>,
+    cert: MaybeMultiple<CertInput>,
+    ca?: MaybeMultiple<CertInput>,
+    ciphers?: MaybeMultiple<string>,
+    honorCipherOrder?: boolean
+  };
+
+  type HttpsSettings = 
+    & CustomHttpsSettings
```

Comment:
Should there be something before the `&` or is this a typescript syntax that I'm not familiar with?

Reply:
It's something TypeScript supports. `|` as well. It's just a syntax thing that lets you do a list of unions/intersections in a nicer-looking way.

Reply:
Ok, that's fine. I'm just surprised that a leading `&` is allowed. I'd have expected something more like:

```ts
type HttpsSettings =
  CustomHttpsSettings &
  Omit<
     SecureContextOptions,
     keyof CustomHttpsSettings | 'secureProtocol' | 'secureOptions'
     >;
```

Reply:
Perhaps we can lint on code styles here ([see comment](https://github.com/godaddy/gasket/pull/311#discussion_r769998926)).